File: handlers/FCMHandler.go
package handlers

import (
    "net/http"
    "encoding/json"
    "ur-admin-backend/models"
    "ur-admin-backend/services"
    "strings"
)

func FCMHandler(w http.ResponseWriter, r *http.Request){
    if err := CheckLogin(r); err != nil {
      http.Error(w, err.Error(), http.StatusUnauthorized)   
        return
    }
    var msg models.FCMMessage
    if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
        http.Error(w, "Bad request", http.StatusBadRequest)
        return
    }

    response, err := services.SendFCMMessage(msg)
    if err != nil {
        http.Error(w, "Error enviando mensaje", http.StatusInternalServerError)
        return
    }
    if strings.Contains(response, "enviado") == false {
        w.WriteHeader(http.StatusInternalServerError)
        w.Write([]byte("{\"code\":500, \"error\":\"message not sent\"}"))
        return
    }
    w.WriteHeader(http.StatusOK)
    w.Write([]byte("{\"code\":200, \"description\":\""+ response+"\"}"))
}

File: handlers/fcmHandler.js
const { sendFCMMessage } = require('../services/fcmSvc');
const { checkLogin } = require('./loginHandler');

async function fcmHandler(req, res) {
  try {
    await checkLogin(req);

    const msg = req.body;
    if (!msg || !msg.notifications) {
      return res.status(400).send('Bad request');
    }

    const response = await sendFCMMessage(msg);
    if (!response.includes('enviado')) {
      return res.status(500).json({ code: 500, error: 'message not sent' });
    }

    res.status(200).json({ code: 200, description: response });
  } catch (error) {
    if (error.message.includes('Authorization') || error.message.includes('Token') || error.message.includes('Application')) {
      return res.status(401).send(error.message);
    }
    console.error('FCM Error:', error);
    res.status(500).json({ code: 500, error: 'Error enviando mensaje' });
  }
}

module.exports = { fcmHandler };
